controllers/common/common_status_update.go

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) 2023 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	appsv1 "open-cluster-management.io/multicloud-operators-subscription/pkg/apis/apps/placementrule/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	policiesv1beta1 "open-cluster-management.io/governance-policy-propagator/api/v1beta1"
+)
+
+// RootStatusUpdat update Root policy status with bound decisions and placements
+func RootStatusUpdate(c client.Client, rootPolicy *policiesv1.Policy) error {
```

Comment:
I think you should have a context as an argument since you'll have a context from the event handler or reconcile request that can be shared with the API (actually uses the cache) requests in calling functions.